NBitcoin/IpExtensions.cs

```diff
@@ -459,6 +530,13 @@ public static bool IsMulticast(this IPAddress address)
 				   || (bytes[15 - 15] == 0xFF);
 		}
 
+		public static bool IsRoutable(this EndPoint endpoint, bool allowLocal)
+		{
+			if (endpoint is IPEndPoint ip)
+				return ip.Address.IsRoutable(allowLocal);
+
+			return true; 

```

Comment:
Shouldn't be the fallback return false - it would be safer and investigate all the valid cases and return true only there?

Reply:
No, otherwise Tor v3 should be no routable.

Reply:
ACK